packages/flutter_tools/test/integration/test_driver.dart

```diff
@@ -125,66 +137,85 @@ abstract class FlutterTestDriver {
     return isolate;
   }
 
+  Future<void> breakAt(Uri uri, int line) async {
+    await addBreakpoint(uri, line);
+    await waitForPause();
```

Comment:
I don't know if it matters for any of the tests we have, but this code looks racy if the code with the breakpoint only runs once. Is it possible the code for the breakpoint has already run by the time the breakpoint is added, so it never pauses?

(the original code used reloads/restarts to avoid this, but it was kinda clumsy!)